python/ray/serve/deployment.py

```diff
@@ -466,7 +462,11 @@ def deployment_to_schema(d: Deployment) -> DeploymentSchema:
         ),
         init_args=(),
         init_kwargs={},
-        num_replicas=d.num_replicas,
+        # TODO(Sihan) DeploymentConfig num_replicas and auto_config can be set together
+        # because internally we use these two field for autoscale and deploy.
+        # We can improve the code after we separate the user faced deployment config and
+        # internal deployment config.
+        num_replicas=None if d._config.autoscaling_config else d.num_replicas,
```

Comment:
@shrekris-anyscale is this okay? asking you since you recently touched these config path

Reply:
Yes, this should be okay.